### ISSUE TITLE
Synchronization

### ISSUE THREAD

**Issue by ldelmont** [hcrest/sh2-demo-nucleo#3]
I have activated SH2_ACCELEROMETER, SH2_GYROSCOPE_CALIBRATED and SH2_MAGNETIC_FIELD_CALIBRATED at the same report Interval. i am receiving almost all the time Accelerometer data. Every 5 or 6 accelerometer data i am receiving one gyroscope or mangnetomer data.

E.g: Acc, Acc, Acc, Mag, Acc, Acc, Acc ,Acc, Gyr, Acc, Acc ,Acc,

 What should i do to receive one data of each sensor ordered.
E.g: Acc, Mag, Gyros, Acc, Mag, Gyros, Acc, Mag, Gyros

Thank you.

**Comment by DaveOfThings** [comment]
What report interval did you request?  The sensors only support certain discrete rates.  When you request an arbitrary interval, the sensor is configured for the best supported rate.  This is usually the next higher rate than what you requested or the highest available rate.

The problem could also be that your communications channel can't support the number of sensor reports requested.  This may happen when using i2c, for example, and enabling several sensors at a high rate.  The I2C bus simply can't carry all the requested reports and some are dropped.

**Comment by ldelmont** [comment]
Yes i am using i2c but the request is every 100ms for the 3 sensors.

**Comment by DaveOfThings** [comment]
100ms (10Hz) is a very modest data rate, I wouldn't expect any problems with that.  Is it possible you accidentally configured the sensors for a 100us interval?  Do you have any code I could look at or any logic analyzer traces?

**Comment by ldelmont** [comment]
With the same code  sh2-demo-nucleo of the repository I have the same problem.

Configuration:
[cid:image002.png@01D572F8.B0BA6880]

Terminal :

[cid:image001.png@01D572F9.14A2E2A0]

Interruptions: sometimes I have 100 ms between the same sensor interrupt and sometimes no.

[cid:image003.png@01D572F9.E09BD5B0]


In the case of my personal project I think that the result is wosrt because I am working with a cortex m0 (stm32l010).

De: David Wheeler <notifications@github.com>
Enviado el: Tuesday, September 24, 2019 3:57 PM
Para: hcrest/sh2-demo-nucleo <sh2-demo-nucleo@noreply.github.com>
CC: Lucas Adrián Delmonte <ldelmonte@leitat.org>; Author <author@noreply.github.com>
Asunto: Re: [hcrest/sh2-demo-nucleo] Synchronization (#3)


100ms (10Hz) is a very modest data rate, I wouldn't expect any problems with that. Is it possible you accidentally configured the sensors for a 100us interval? Do you have any code I could look at or any logic analyzer traces?

**Comment by DaveOfThings** [comment]
The images you references aren't visible.  I just see URLs starting with "cid:"

**Comment by ldelmont** [comment]
With the same code  sh2-demo-nucleo of the repository I have the same problem.

Configuration:
![code](https://user-images.githubusercontent.com/51952648/65527164-8537f680-def2-11e9-8725-58def455252f.png)

Terminal :
![terminal](https://user-images.githubusercontent.com/51952648/65527210-97b23000-def2-11e9-8c95-aaf084d86eb1.png)

Interruptions: sometimes I have 100 ms between the same sensor interrupt and sometimes no.
![scope_1](https://user-images.githubusercontent.com/51952648/65527255-a8fb3c80-def2-11e9-8d0a-10d1d5fa44d4.png)

In the case of my personal project I think that the result is worst because I am working with a cortex m0 (stm32l010).
